tools/blue_deployment/step1.js

```diff
@@ -0,0 +1,343 @@
+ *
+ * @returns {ChainInfo} The chain information of the most recent block on the chain.
+ */
+const getChainInfo = async (rpcEndpoint) => {
+    const web3 = new Web3(rpcEndpoint);
+
+    const chainId = await web3.eth.net.getId();
+    const block = await web3.eth.getBlock('latest');
+
+    return {
+        chainId,
+        blockHeight: block.number,
+        stateRoot: block.stateRoot,
+    };
+};
+
+/**
+ * Create Contract instances for libraries MerklePatriciaProof, GatewayLib, MessageBus.
+ *
+ * @returns {Object.<string, Contract>} Map from contract names to contract instances.
+ */
+const createLibraryConracts = () => {
+    const MerklePatriciaProof = Contract.loadTruffleContract('MerklePatriciaProof', null, { rootDir });
+    const GatewayLib = Contract.loadTruffleContract('GatewayLib', null, { rootDir });
+    GatewayLib.addLinkedDependency(MerklePatriciaProof);
+    const MessageBus = Contract.loadTruffleContract('MessageBus', null, { rootDir });
+    MessageBus.addLinkedDependency(MerklePatriciaProof);
+
+    return {
+        MerklePatriciaProof,
+        GatewayLib,
+        MessageBus,
+    };
+};
+
+/**
+ * Create an Organization Contract
+ *
+ * @param {Address} adminAddress Address of the organization admin.
+ * @param {Address} ownerAddress Address of the organization owner.
+ * @param {Array.<Address>} workerAddresses Addresses of the organization workers.
+ * @param {expirationHeight} expirationHeight Block height at which the
+ *                           workers will expire.
+ *
+ * @returns {Contract} The Organization contract instance.
+ */
+const createOrganization = (adminAddress, ownerAddress, workerAddresses, expirationHeight) =>  {
+    return Contract.loadTruffleContract('Organization', [
+        adminAddress,
+        ownerAddress,
+        workerAddresses,
+        expirationHeight,
+    ], { rootDir });
+};
+
+/**
+ * Create an Anchor Contract
+ *
+ * @param {Address|ContractReference} organization The address of or reference
+ *                                    to the Organization contract for the anchor.
+ * @param {string} remoteChainId The chain id of the remote chain to anchor.
+ * @param {number} remoteBlockHeight The block height of the remote chain to anchor.
+ * @param {string} remoteStateRoot The state root of the remote chain to anchor.
+ * @param {number} maxStateroots The max number of state roots to store in
+ *                               thecircular buffer.
+ *
+ * @returns {Contract} The Anchor contract instance.
+ */
+const createAnchor = (
+    organization,
+    remoteChainId,
+    remoteBlockHeight,
+    remoteStateRoot,
+    maxStateroots = '10',
+) => {
+    return Contract.loadTruffleContract(
+        'Anchor',
+        [
+            remoteChainId,
+            remoteBlockHeight,
+            remoteStateRoot,
+            maxStateroots,
+            organization,
+        ],
+        { rootDir },
+    );
+};
+
+const createEIP20Gateway = (
```

Comment:
Should all the contract specific create and deploy functions be defined here? It could make sense to put that code into files, one per contract, and call them when deploying. Sounds cleaner to me.

Reply:
I think one per contract is a bit excessive, but they should be moved into a separate file.

Reply:
👍